edx-solutions/edx-platform
# (WIP) YONK-529 Discussions web fragments prototype

Branches from `open-craft:carlio/OC-2714-web-fragments2`, which is not merged to master yet, hence the big diff.

This is a WIP attempt for allowing the discussion tab in Apros to be rendered via web fragments, so that the course discussion XBlock is not involved anymore. It makes use of changes to `jquery-xblock-2.js` in the Apros repository, in order to display web fragments similarly to XBlocks.

The difficult part is proving to be making RequireJS work both on the standalone view and via `jquery-xblock`, specifically the require/define imports while adding JS files and inline JS blocks dynamically.

**Testing instructions**:
1. Apply [this patch](https://github.com/edx-solutions/edx-platform/files/1308342/bogdan-jquery-xblock-fragments.diff.zip) to Apros's `jquery-xblock` that allows web fragments to be displayed via `jquery-xblock`. Please note that that's also WIP and it's likely it will need more work.
2. Add a section named `DISCUSSION_TAB` to a course and in Apros go to the Discussions tab. 
3. Check the JS console for the import error messages.

**Reviewers**:
